### ISSUE TITLE
Security concern

### ISSUE THREAD

**Issue by jbrepogmailcom** [klausw/hackerskeyboard#801]
Hello. I understand that any Android keyboard passes keystrokes to the operating system over some API, where it is then picked up by target application over another API. Does that mean that on level of OS any software keyboard can be eavesdropped? Thank you for comment. Jan

**Comment by lbdroid** [comment]
Your question is very naive.
At OS level, EVERYTHING can be eavesdropped.

**Comment by 5c0tt-b0t** [comment]
Yes, of course, but if you mean... "Can other *apps* on my phone read what my keyboard is typing" then no, normal apps without root access wouldnt be able to, if you mean... "Can other apps read my *clipboard*" then yes (the android system clipboard has nothing to do with this keyboard app though) though it depends on what version of Android you use, Android 11 has stopped this behaviour so apps cannot anymore read the clipboard content.

**Comment by 5TDg9** [comment]
They still can (and sometimes must) read the Clipboard even with Android 11... BUT they must run in foreground.
So ne carefull and remember to always clear/overwrite your sensitive clipboard content before using a shady app.
